New Entry: R CSV files + term entries (#883)
Codecademy/docs: content/r/concepts/data-frames/terms/ncol/ncol.md

---
Title: 'ncol()'
Description: 'Returns the number of columns in a data frame.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'

>   - 'Data Structures'

Reviewer: ```suggestion
  - 'Data Structures'
  - 'Methods'
  - 'Tables'
```

CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

The **`ncol()`** function returns the number of columns in a data frame.

## Syntax

```pseudo
ncol(df)
```

- `df`: The data frame to call the function on.

## Example

> Suppose there exists a `patients.csv` file with the following information:

Reviewer: ```suggestion
The following example uses a  [CSV file](https://www.codecademy.com/resources/docs/r/csv-files) named `"patients.csv"` file with the following information:
```

I noticed this from the `colnames.md` file, but should we surround in backticks because it's a file name yet keep the quotes because it's passed into `read.csv()` as a string?

| First Name | Last Name | DOB        |
| ---------- | --------- | ---------- |
| Kiercen    | Tremboly  | 06/24/1992 |
| Phaedra    | Blackborn | 08/19/1972 |
| Toby       | Spencer   | 11/16/1953 |
| Sandra     | Speller   | 07/05/1989 |

The data within the [CSV file](https://www.codecademy.com/resources/docs/r/data-types) can be read in and the `ncol()` function can be used to retrieve the number of columns:

```r

> # Read in file

Reviewer 2: ```suggestion
# Read file
```

---
Title: 'ncol()'
Description: 'Returns the number of columns in a data frame.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Data Structures'
  - 'Methods'
  - 'Tables'
CatalogContent:
  - 'learn-r'
  - 'paths/computer-science'
---

The **`ncol()`** function returns the number of columns in a data frame.

## Syntax

```pseudo

> ncol(data_frame)

Reviewer: ```suggestion
ncol(df_object)
```

```

> The `data_frame` parameter is the data frame passed into the `ncol()` function.

Reviewer: ```suggestion
The `df_object` parameter is the data frame passed into the `ncol()` function.
```
